### DISCUSSION TITLE
Browser refresh for info pages

### DISCUSSION THREAD

**Discussion by douglaspalmer** [keycloak/keycloak#14377] [Feedback]
While working on #13922 I discovered that the problem is broader than described in the issue. If an information page is refreshed then the same error message is displayed. This is because the authentication session is ended when the information page is displayed.

Marek and I have been thinking about potential solutions to this problem. One suggestion is to keep the session alive for a minute after navigating to the information page. Is this a good idea? If not then does anyone have a better solution?

**Comment by stianst** [comment]
Problem with that is it works for a few minutes, but not if it's a little bit later for example someone goes and grabs a coffee and comes back.

Some other ideas maybe:

* Use browser cache headers to prevent browser from fetching the page again from KC
* Just display some more friendly page expired message of sorts instead of a error page

**Reply by douglaspalmer** [reply]
If we use a cache header should we also remove the locale selection box from the info pages?

**Reply by mposolda** [reply]
The "Browser cache headers solution" will work for the case when user refreshes the browser page. However it won't work when user is switching locale. Also I don't know if it has any other side-effects? For example assume that the info page is shown with the URL "http://localhost:8081/auth/realms/test/login-actions/required-action" . Not 100% sure if it can cause returning of the cached page when the same URL is displayed for the user next time when user does any action? This would be incorrect as next action should not display cached page.

So right now, I am more towards to:
- Hide the locale selector from the "info" pages in case that authenticationSession was just removed. Will it be bad for usability? In my opinion, not big problem as the "info" page is displayed after some other page and hence user already had a chance to change the locale on the previous page?
- Show some "Action expired" page when there is page refresh. (in other words, when there is no valid authenticationSession). This will be more friendly than current "We are sorry. Cookie not found", which is clearly incorrect.
